File: advanced1/cpp/no1316.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

bool isGroupWord(const string &word) {
    vector<bool> alphabet(26, false);
    char prev = '\0';

    for (char c : word) {
        if (c != prev) {
            if (alphabet[c - 'a']) {
                return false;
            }
            alphabet[c - 'a'] = true;
        }
        prev = c;
    }
    return true;

}

int main() {
    int N;
    cin >> N;

    int groupWordCount = 0;

    for (int i = 0; i < N; ++i) {
        string word;
        cin >> word;
        if (isGroupWord(word)) {
            groupWordCount++;
        }
    }
    cout << groupWordCount << endl; // 그룹 단어의 개수 출력
    
    return 0;
}

File: advanced1/python/no1316.py
def isGroupWord(word):
    alphabet = [False] * 26
    prev = ''

    for c in word:
        if (c != prev):
            if (alphabet[ord(c) - ord('a')]):
                return False
            alphabet[ord(c) - ord('a')] = True
        prev = c
    return True

def main():
    N = int(input())
    groupWordCount = 0
    i = 0
    while(i < N):
        word = input()
        if (isGroupWord(word)):
            groupWordCount += 1
        i += 1

    print(groupWordCount)

main()